### ISSUE TITLE
is there a callback if a transition fails?

### ISSUE THREAD

**Issue by siong1987** [aasm/aasm#357]
i am poking around with the code to find a possible callback that i can use if a transition fails.

it seems like i could use `aasm_event_failed` for it but it gets triggered for all failed transition events.  that means that i need to implement the logic to check what the event fails and handle it accordingly.

unless i am missing something obvious in the code, can someone point me to it?

**Comment by t3hk0d3** [comment]
What do you mean under "fails"? Raises exception?
There is ``error`` callback for event.

```ruby
event :boo, error: [:on_error] do
 ....
end
```

Unfortunately there is no such callback per-transition.

**Comment by siong1987** [comment]
@t3hk0d3 for example, when a state isn't supposed to be transitioned to another state.

`on_error` seems to only work for exception.

**Comment by t3hk0d3** [comment]
@siong1987 I might be wrong, but looks like ``on_error`` capture ``AASM::InvalidTransition`` as well.

**Comment by siong1987** [comment]
yeah, but not when `whiny_transitions` is set to false :(

**Comment by t3hk0d3** [comment]
@siong1987 Well you can mess around on higher-level, and perform ``may_<event>?`` manually. It would be effective workaround imho.

**Comment by siong1987** [comment]
@t3hk0d3 yeah, understand the workaround.  but just feel like it should be a callback that should happen.
